packages/material-ui-lab/src/TreeItem/TreeItem.js

```diff
@@ -16,38 +16,48 @@ export const styles = theme => ({
     padding: 0,
     outline: 0,
     WebkitTapHighlightColor: 'transparent',
-    '&:focus > $content': {
-      backgroundColor: theme.palette.grey[400],
+    '&:focus > $content $label': {
+      backgroundColor: theme.palette.action.hover,
+    },
+    '&$selected > $content $label': {
+      backgroundColor: theme.palette.action.selected,
```

Comment:
I recall a discussion where we were wondering about the right opacity or colors for the different states. I have recently discovered https://material.io/design/interaction/states.html#anatomy. We should be able to leverage it in this component :). While we don't follow it everywhere yet, we have started to. It could be handled in a follow-up pull request.

However, it's not clear how multiple states should interact, e.g. selected + hovered.

Reply:
Pretty sure that was part of the previous discussion.

The overlay for focus only works when you have adjacent components of the same type that are not focused to visually compare with.

Used isolation, for example with a single button, it isn't possible to tell with an overlay whether the component is focused, or if that is just its default color. Contrast that with the focus ripple, which, while not altogether attractive, does serve a purpose.

So – good for here, but not necessarily for every component.

Reply:
Wouldn't moving to the overlay method break existing expectation with the theme system?

I still wouldn't mind looking into this in the future :)